packages/ui/src/providers/ListQuery/index.tsx

```diff
@@ -165,7 +165,13 @@ export const ListQueryProvider: React.FC<ListQueryProps> = ({
 
       hasSetInitialParams.current = true
     }
-  }, [defaultSort, defaultLimit, router, modifySearchParams, currentQuery])
+
+    // if there are no docs on the current page and docs exist, redirect to the last page
+    if (data.docs.length === 0 && data.totalDocs > 0) {
+      currentQuery.page = String(data.totalPages)
```

Comment:
I think it might make more sense to go back to page 1, no?

Reply:
Yeah you're right, no value in taking them to the closest page. Updated!